src/components/calcite-label/calcite-label.tsx

```diff
@@ -74,24 +74,72 @@ export class CalciteLabel {
   //--------------------------------------------------------------------------
 
   @Listen("click")
-  onClick({ target }: MouseEvent): void {
-    if (target === this.el || target === this.labelEl || target === this.spanEl) {
-      const forAttr = this.el.getAttribute("for");
-      this.calciteLabelFocus.emit({
-        labelEl: this.el,
-        interactedEl: target as HTMLElement,
-        requestedInput: forAttr
-      });
-      const inputForThisLabel: HTMLElement = forAttr
-        ? document.getElementById(forAttr)
-        : this.el.querySelector("input");
-      if (
-        (inputForThisLabel && inputForThisLabel.nodeName.startsWith("CALCITE-")) ||
-        (inputForThisLabel && inputForThisLabel.nodeName === "INPUT" && target === this.el)
-      ) {
-        inputForThisLabel.click();
+  onClick(event: MouseEvent): void {
+    const target = event.target as HTMLElement;
+    this.calciteLabelFocus.emit({
+      labelEl: this.el,
+      interactedEl: target,
+      requestedInput: this.for
+    });
+
+    // 1. has htmlFor
+    if (!this.for) return;
+
+    // 2. htmlFor matches a calcite component
+    const inputForThisLabel = document.getElementById(this.for);
+    if (!inputForThisLabel) return;
+    if (!inputForThisLabel.localName.startsWith("calcite")) return;
+
+    // 5. target is NOT the calcite component that this label matches
+    if (target === inputForThisLabel) return;
+
+    // 3. target is not a labelable native form element
+    const labelableNativeElements = [
+      "button",
+      "input",
+      "meter",
+      "output",
+      "progress",
+      "select",
+      "textarea"
+    ];
+    if (labelableNativeElements.includes(target.localName)) return;
+
+    // 4. target is not a labelable calcite form element
+    const labelableCalciteElements = [
+      "calcite-button",
+      "calcite-checkbox",
+      "calcite-date",
+      "calcite-inline-editable",
+      "calcite-input",
+      "calcite-radio",
+      "calcite-radio-button",
+      "calcite-radio-button-group",
+      "calcite-radio-group",
+      "calcite-rating",
+      "calcite-select",
+      "calcite-slider",
+      "calcite-switch"
+    ];
+    if (labelableCalciteElements.includes(target.localName)) return;
+
+    // 5. target is not a child of a labelable calcite form element
+    for (let i = 0; i < labelableCalciteElements.length; i++) {
+      if (target.closest(labelableCalciteElements[i])) {
+        return;
       }
     }
+
+    const nativeInputForThisLabel: HTMLElement = inputForThisLabel.querySelector(
+      `#${this.for}-input`
+    );
+    if (nativeInputForThisLabel) {
```

Comment:
nitpick:

```
const inputToFocus = nativeInputForThisLabel || inputForThisLabel;
inputToFocus?.click();
```

Reply:
I removed this check altogether as it was redundant since clicks on the host element accomplish the same thing as clicking the hidden native input on some components.  Even simpler now!